crates/turbo-tasks-memory/src/task.rs

```diff
@@ -1058,6 +1083,7 @@ impl Task {
                                 format!("TaskState({})::event", description())
                             }),
                             outdated_edges: Box::new(outdated_edges),
+                            clean: recomputing,
```

Comment:
Should this not be clean if it is not currently recomputing?

Reply:
If it's marked as dirty due to changes, it's not clean.
If it's marked as dirty when trying to read a missing value, it's still clean.

Reply:
I split mark_dirty_internal into two methods: mark_dirty and recompute to make it more clear.